### ISSUE TITLE
Error when trying to add VB analyzer created via Roslyn SDK templates as a nuget reference

### ISSUE THREAD

**Issue by shyamnamboodiripad** [dotnet/roslyn#1596]
I see the below error for the above scenario -

![capture](https://cloud.githubusercontent.com/assets/10579684/6838029/c620e6a4-d310-11e4-80ff-ce8d499fb856.PNG)

The corresponding rule has no title text when I look at it under solution explorer -
![capture2](https://cloud.githubusercontent.com/assets/10579684/6838083/261b2de4-d311-11e4-82dc-3a9de15a4b54.PNG)

And I see warnings with empty descriptions in the error list -
![capture3](https://cloud.githubusercontent.com/assets/10579684/6838129/d4ee10c0-d311-11e4-8ff0-9e26105d28a6.PNG)

The VB analyzer project created using template has the following code to initialize the title that should be showing up above -

Friend Shared ReadOnly Title As LocalizableString = New LocalizableResourceString(NameOf(My.Resources.AnalyzerTitle), My.Resources.ResourceManager, GetType(My.Resources.Resources))

Note that I haven't tried to do anything out of the way like change the UI culture / tried to localize the analyzer. I don't seem to run into this issue for identical C# analyzer created using the template.

**Comment by mavasani** [comment]
@shyamnamboodiripad this seems to be an issue with the _"Resources.Designer.vb"_ file checked into our VB analyzer template.

This file has the following line that hardcodes _"My.Resources.Resources"_ for the first argument to ResourceManager constructor:

```vb
Dim temp As Global.System.Resources.ResourceManager = New Global.System.Resources.ResourceManager("My.Resources.Resources", GetType(Resources).GetTypeInfo.Assembly)
```

This should instead have a token substitution for the project name, just like the C# template's _"Resources.Designer.cs"_ has:

```
global::System.Resources.ResourceManager temp = new global::System.Resources.ResourceManager("$saferootprojectname$.Resources", typeof(Resources).GetTypeInfo().Assembly);
```

NOTE: There is a workaround for the current VB analyzer project. Just right click on the _"Resources.resx"_ file in your project under _"My Project"_ node and execute _"Run Custom Tool"_ command. This should regenerate the correct designer file and everything should work fine.

@jmarolf I am moving this bug over to you :)

**Comment by shyamnamboodiripad** [comment]
Looks like fixing this would also fix #1398

**Comment by shyamnamboodiripad** [comment]
This is now fixed (sources are for SDK templates are currently internal).

@jmarolf Could you please verify this.
